src/environmentVariables.ts

```diff
@@ -0,0 +1,147 @@
+        if (typeof key === 'string') {
+            let needSet = true;
+            if (value === undefined) {
+                needSet = false;
+            } else if (value === null) {
+                if (!this.preserveNull) {
+                    needSet = false;
+                }
+            } else if (typeof value !== 'string') {
+                value = '' + value;
+            }
+            if (needSet) {
+                const existKey = this.getKey(key, true);
+                return Reflect.set(this[envProperty], existKey, value, receiver);
+            } else {
+                return Reflect.deleteProperty(this[envProperty], key);
+            }
+        }
+        return false;
+    }
+
+    [inspect]() {
+        return util.inspect(this[envProperty]);
+    }
+
+    toString() {
+        return this[envProperty].toString();
+    }
+}
+
+export class EnvironmentVariablesUtils {
+
+    public static create(from?: Map<string, string> | EnvironmentVariables | null, preserveNull?: boolean, isWin32?: boolean): EnvironmentVariablesUndefined {
+        const env = new EnvironmentVariablesPrivate(preserveNull, isWin32);
+        const p = new Proxy(env, {
+            defineProperty: (target, p, attributes) => Reflect.defineProperty(target[envProperty], p, attributes),
+            deleteProperty: (target, p) => Reflect.deleteProperty(target[envProperty], p),
+            get: (target, p) => {
+                if (typeof p === 'string') {
+                    return target.get(p);
+                }
+                return Reflect.get(target, p);
+            },
+            getOwnPropertyDescriptor: (target, p) => {
+                if (typeof p === 'string') {
+                    return Reflect.getOwnPropertyDescriptor(target[envProperty], target.getKey(p, false));
+                } else {
+                    return Reflect.getOwnPropertyDescriptor(target, p);
+                }
+            }
+            ,
+            has: (target, p) => Reflect.has(target[envProperty], p),
+            ownKeys: (target) => Reflect.ownKeys(target[envProperty]),
+            set: (target, p, value, receiver): boolean => target.set(p, value, receiver)
+        }) as unknown as EnvironmentVariablesUndefined;
+        if (from !== undefined && from !== null) {
+            if (from instanceof Map) {
+                for (const [key, value] of from.entries()) {
+                    p[key] = value;
+                }
+            } else {
+                Object.assign(p, from);
+            }
+        }
+        return p;
+    }
+
+    public static createPreserveNull(from?: Map<string, string> | EnvironmentVariables | null): EnvironmentVariables {
+        return EnvironmentVariablesUtils.create(from, true);
+    }
+
+    public static mergeImpl(preserveNull: boolean, isWin32: boolean | undefined, ...envs: EnvironmentVariablesItem[]): EnvironmentVariablesUndefined {
```

Comment:
An `impl` method shouldn't be public.  Can we just have:
```ts
public static merge(options: EnvironmentOptions, ...envs: Environment[]): Environment
```
You can remove the other convenience methods.